src/app/services/geo-data.service.ts

```diff
@@ -100,6 +109,46 @@ export class GeoDataService {
     );
   }
 
+  setFeatures(fc: FeatureCollection) {
+    fc.features = fc.features.map((feat: Feature) => new Feature(feat));
+    this.getGroups(fc.features);
+    this._features.next(fc);
+  }
+
+  setFeatureProperties(featureId: string | number, properties: any) {
+    this.features.pipe(first()).subscribe((fc) => {
+      fc.features = fc.features.map((feat: Feature) => {
+        if (feat.id === featureId) {
+          feat.properties = properties;
+        }
+        return feat;
+      });
+      this.setFeatures(fc);
+    });
+  }
+
+  updateFeatureTags(projectId) {
+    this._updatedTagFeatures.value.forEach((featureId) => {
```

Comment:
1) should we add a docstr here like about persisting the queued tag features to the back end.
2) is there another name we can use for `_updatedTagFeatures`.  I'm not certain but _toBePersistedChangedTagFeatures or _qeuedTagFeatures?  not certain but just wondering if it can be more descriptive.

Reply:
Changed the wording to `_tagFeaturesQueue` and added a note above `setTagFeaturesQueue`.